lib/members_page.rb

```diff
@@ -12,28 +12,50 @@ class MembersPage < Scraped::HTML
     party = party_id = nil
     table.xpath('.//tr').map do |tr|
       tds = tr.css('td')
-      if tds.size == 1
-        img = tr.at_css('img')
-        next if img.nil? || img[:party_name].to_s.empty?
-        party = img[:party_name]
-        party_id = img[:party_id]
-        next
+      type = type_of_row(tds)
+      if type == :party_image
+        party, party_id = party_image(tds, party, party_id)
       end
-      next if tds[1].text == 'Entidad'
-
-      area, area_id = ocd(tds[1].text.tidy, tds[2].text.tidy)
-
-      (fragment tds => MemberRow).to_h.merge(
-        party:    party,
-        party_id: party_id,
-        area_id:  area_id,
-        area:     area
-      )
+      next if type == :filler || type == :party_image
+      member_row(tds, party, party_id)
     end.reject(&:nil?)
   end
 
   private
 
+  def member_row(row, party, party_id)
```

Comment:
I'm afraid that this commit is making things substantially worse, rather than better. You're really just creating a big bowl of spaghetti functions, rather than splitting out simple methods. Having methods that take parameters here is a giant red flag. Occasionally we allow pre-existing simple convenience functions to slip through in the name of expediency, but you shouldn't be creating new ones. You're operating in a class-based environment here, so if you're passing things like 'row' around (here and below), it implies that should be on a Row-based class, not on the full class.